apiary.apib

```diff
@@ -398,11 +398,318 @@ MerkleDAG.
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 21 Jan 2016 17:09:00 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Body
 
-## unwant
+        ```
+        {
+          "ProvideBufLen": 0,
+          "Wantlist": null,
+          "Peers": [
+            "QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+            "QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z",
+            "QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm"
+          ],
+          "BlocksReceived": 0,
+          "DupBlksReceived": 0,
+          "DupDataReceived": 0
+        }
+        ```
+
+## unwant [GET /bitswap/unwant{?arg}]
+Remove a given block from your wantlist
+
++ Parameters
+    + arg (Multihash, required) - key to remove from your wantlist
+
++ Request No Arguments
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/unwant
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/unwant
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Fri, 29 Jan 2016 20:25:10 GMT
+        Content-Length: 26
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Body
 
-## wantlist
+        ```
+        Argument 'key' is required
+        ```
+
++ Request Empty Argument
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=
+        ```
+
++ Response 500 (application/json)
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 29 Jan 2016 20:40:32 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: 'incorrectly formatted key: '
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "incorrectly formatted key: ",
+          "Code": 0
+        }
+        ```
+
++ Request With Key
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ
+        ```
+
++ Response 200 (application/json)
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 29 Jan 2016 20:35:58 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Body
+
+        ```
+        ```
+
+## wantlist [GET /bitswap/wantlist{?peer}]
+Show blocks currently on the wantlist.
```

Comment:
because if the param 'peer' is passed, we will see the want list that we have from that peer, if we don't pass anything, our own want list will be shown